frontend/app/src/views/defi/Airdrops.vue

```diff
@@ -0,0 +1,161 @@
+            :footer-props="footerProps"
+            :headers="headers"
+          >
+            <template #item.address="{ item }">
+              <hash-link :text="item.address" />
+            </template>
+            <template #item.amount="{ item }">
+              <amount-display :value="item.amount" :asset="item.asset" />
+            </template>
+            <template #item.source="{ item }">
+              <div class="d-flex flex-row align-center">
+                <v-img
+                  width="24px"
+                  contain
+                  position="left"
+                  max-height="32px"
+                  max-width="32px"
+                  :src="getIcon(item.source)"
+                />
+                <span class="ms-2" v-text="getLabel(item.source)" />
+              </div>
+            </template>
+          </v-data-table>
+        </v-card-text>
+      </v-card>
+    </v-container>
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator';
+import { DataTableHeader } from 'vuetify';
+import { mapActions, mapGetters } from 'vuex';
+import ProgressScreen from '@/components/helper/ProgressScreen.vue';
+import { footerProps } from '@/config/datatable.common';
+import StatusMixin from '@/mixins/status-mixin';
+import { Section } from '@/store/const';
+import {
+  AIRDROP_1INCH,
+  AIRDROP_TORNADO,
+  AIRDROP_UNISWAP
+} from '@/store/defi/const';
+import { Airdrop, AirdropType } from '@/store/defi/types';
+import { ETH, GeneralAccount } from '@/typing/types';
+
+type AirdropSource = {
+  readonly icon: string;
+  readonly name: string;
+};
+
+type AirdropSources = {
+  readonly [source in AirdropType]: AirdropSource;
+};
+
+@Component({
+  components: { ProgressScreen },
+  computed: {
+    ...mapGetters('defi', ['airdrops', 'airdropAddresses'])
+  },
+  methods: {
+    ...mapActions('defi', ['fetchAirdrops'])
+  }
+})
+export default class Airdrops extends Mixins(StatusMixin) {
+  readonly section = Section.DEFI_AIRDROPS;
+  readonly ETH = ETH;
+  readonly footerProps = footerProps;
+  readonly headers: DataTableHeader[] = [
+    {
+      text: this.$t('airdrops.headers.source').toString(),
+      value: 'source'
+    },
+    {
+      text: this.$t('airdrops.headers.address').toString(),
+      value: 'address'
+    },
+    {
+      text: this.$t('airdrops.headers.amount').toString(),
+      value: 'amount',
+      align: 'end'
+    }
+  ];
+  airdrops!: (addresses: string[]) => Airdrop[];
+  fetchAirdrops!: (refresh: boolean) => Promise<void>;
+  selectedAccounts: GeneralAccount[] = [];
+  airdropAddresses!: string[];
+
+  async mounted() {
+    await this.fetchAirdrops(false);
+  }
+
+  get entries(): Airdrop[] {
+    const addresses = this.selectedAccounts.map(({ address }) => address);
+    return this.airdrops(addresses);
+  }
+
+  async refresh() {
+    await this.fetchAirdrops(true);
+  }
+
+  readonly sources: AirdropSources = {
+    [AIRDROP_UNISWAP]: {
```

Comment:
So when we add more, we need to edit those? Need to find a way to keep those in sync in between frontend and backend. Just like with the DeFi protocols too.

Reply:
Well basically this is what we do for the defi protocols too :P

Reply:
yeah I know but we need to figure out a better way for all these stuff to be able to make changes in one place, or at least detect when they fall out of sync. Else it won't scale. Can you spend some thinking juice on it? Probably the test way would make sense?

Reply:
Well one way would be to move this metadata and icon serving to the backend. no need to maintain it twice then.

Reply:
As discussed in the standup can you make an issue for this @kelsos? PErhaps you can also pick it up but we will see. This is not for v1.11.0 though!